engine/apps/api/serializers/direct_paging.py

```diff
@@ -44,21 +46,24 @@ class DirectPagingSerializer(serializers.Serializer):
     title = serializers.CharField(required=False, default=None)
     message = serializers.CharField(required=False, default=None, allow_null=True)
     source_url = serializers.URLField(required=False, default=None, allow_null=True)
-    grafana_incident_id = serializers.CharField(required=False, default=None, allow_null=True)
 
     def validate(self, attrs):
         organization = self.context["organization"]
         alert_group_id = attrs["alert_group_id"]
         title = attrs["title"]
         message = attrs["message"]
         source_url = attrs["source_url"]
-        grafana_incident_id = attrs["grafana_incident_id"]
+        grafana_incident_id = self.ALLOWS_GRAFANA_INCIDENT_ID and attrs.get("grafana_incident_id")
 
         if alert_group_id and (title or message or source_url or grafana_incident_id):
             raise serializers.ValidationError(
-                "alert_group_id and (title, message, source_url, grafana_incident_id) are mutually exclusive"
+                f"alert_group_id and (title, message, source_url{', grafana_incident_id' if self.ALLOWS_GRAFANA_INCIDENT_ID else ''}) "
+                "are mutually exclusive"
             )
 
+        if attrs["users"] and attrs["team"]:
+            raise serializers.ValidationError("users and team are mutually exclusive")
```

Comment:
this is already the case if you perform this workflow from Slack or the web plugin UI, you cannot specify users _and_ a team, we just weren't previously validating the payload for it. Added test cases for this:
- https://github.com/grafana/oncall/pull/4815/files#diff-222ac459a526df2a346ee64684442f734106bc9a5e38209ede4fcf44c0464afcR227-R259
- https://github.com/grafana/oncall/pull/4815/files#diff-7c2413ec1f4ba8bae7008a580d5dc00c7b2224f5182f64624b6879fa9732af05R171-R192